File: src/main/java/mediathek/gui/actions/OpenTargetFolderAction.java
package mediathek.gui.actions;

import mediathek.gui.tabs.tab_downloads.GuiDownloads;
import mediathek.tool.SVGIconUtilities;

import javax.swing.*;
import java.awt.event.ActionEvent;

public class OpenTargetFolderAction extends AbstractAction {
    private final GuiDownloads guiDownloads;

    public OpenTargetFolderAction(GuiDownloads guiDownloads) {
        this.guiDownloads = guiDownloads;
        putValue(Action.NAME, "Zielordner öffnen");
        putValue(Action.SMALL_ICON, SVGIconUtilities.createSVGIcon("icons/fontawesome/folder-open.svg"));
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        guiDownloads.zielordnerOeffnen();
    }
}

File: src/main/kotlin/mediathek/gui/actions/OpenTargetFolderAction.kt
package mediathek.gui.actions

import mediathek.gui.tabs.tab_downloads.GuiDownloads
import mediathek.tool.SVGIconUtilities
import java.awt.event.ActionEvent
import javax.swing.AbstractAction

class OpenTargetFolderAction(private val guiDownloads: GuiDownloads) : AbstractAction() {
    init {
        putValue(NAME, "Zielordner öffnen")
        putValue(SMALL_ICON, SVGIconUtilities.createSVGIcon("icons/fontawesome/folder-open.svg"))
    }

    override fun actionPerformed(e: ActionEvent?) {
        guiDownloads.zielordnerOeffnen()
    }
}
